### PR TITLE
feat(obstacle_cruise_planner): use obstacle velocity based obstacle parameters

### PR DESCRIPTION
**Pull request by danielsanchezaran**
### <samp>🤖 Generated by Copilot at cd523a7</samp>

This pull request enhances the `obstacle_cruise_planner` by adding a new feature to distinguish between moving and static obstacles and adjust the slow down accordingly. It modifies the `planner_interface.hpp` and `planner_interface.cpp` files to implement this feature using new parameters, variables, and functions.
## Related links

[TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-3610)

## Tests performed

On PSim: planned a path with both static and moving obstacles. The example video shows how the vehicle uses different slow down parameters when encountering a moving pedestrian and a static pedestrian (the slow down is more pronounced for the moving pedestrian). A similar test is done with a static and a moving car, in this case, the parked car causes the ego to slow down to around 1 Km/h while the moving car makes the ego slow down to around 7 Km/h. See videos for more details.

https://github.com/autowarefoundation/autoware_launch/assets/25967964/ad3b1786-f2b4-428e-9202-f83adec653fe

https://github.com/autowarefoundation/autoware.universe/assets/25967964/e787ccd3-4797-4828-831d-3fc49c1fb3be

## Notes for reviewers
Note: Documentation update in progress
Requires changes to Autoware launch found on this [PR Draft](https://github.com/autowarefoundation/autoware_launch/pull/681)

## Interface changes

Added parameters for hysteresis-based decision making -> is the obstacle moving or not.

## Effects on system behavior

Vehicle slow down response can be customized for moving or static objects.  

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp [hunk 1]
```diff
@@ -264,34 +268,49 @@ class PlannerInterface
...
+                   {ObjectClassification::BUS, "bus"},
+                   {ObjectClassification::TRAILER, "trailer"},
+                   {ObjectClassification::MOTORCYCLE, "motorcycle"},
+                   {ObjectClassification::BICYCLE, "bicycle"},
+                   {ObjectClassification::PEDESTRIAN, "pedestrian"}};
     }
 
-    ObstacleSpecificParams getObstacleParamByLabel(const ObjectClassification & label_id) const
+    ObstacleSpecificParams getObstacleParamByLabel(
+      const ObjectClassification & label_id, const bool & is_obstacle_moving) const
```
Comment (takayuki5168) [inline planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp:283]: No need to add a reference to bool variables

File: planning/obstacle_cruise_planner/src/planner_interface.cpp [hunk 2]
```diff
@@ -561,9 +561,17 @@ std::vector<TrajectoryPoint> PlannerInterface::generateSlowDownTrajectory(
     const auto & obstacle = obstacles.at(i);
     const auto prev_output = getObjectFromUuid(prev_slow_down_output_, obstacle.uuid);
 
+    bool is_obstacle_moving = [&]() -> bool {
```
Comment (takayuki5168) [inline planning/obstacle_cruise_planner/src/planner_interface.cpp:564]:
```suggestion
    const bool is_obstacle_moving = [&]() -> bool {
```

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp [hunk 3]
```diff
@@ -264,34 +268,49 @@ class PlannerInterface
...
-    ObstacleSpecificParams getObstacleParamByLabel(const ObjectClassification & label_id) const
+    ObstacleSpecificParams getObstacleParamByLabel(
+      const ObjectClassification & label_id, const bool is_obstacle_moving) const
     {
-      const std::string label = types_map.at(label_id.label);
-      if (obstacle_to_param_struct_map.count(label) > 0) {
-        return obstacle_to_param_struct_map.at(label);
-      }
-      return obstacle_to_param_struct_map.at("default");
+      std::string label =
```
Comment (takayuki5168) [inline planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp:285]:
```suggestion
      const std::string label =
```

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp [hunk 4]
```diff
@@ -264,34 +268,49 @@ class PlannerInterface
...
+        (types_map.count(label_id.label) > 0) ? types_map.at(label_id.label) : "default";
+      std::string movement_postfix = (is_obstacle_moving) ? "moving" : "static";
```
Comment (takayuki5168) [inline planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp:287]:
```suggestion
      const std::string movement_postfix = is_obstacle_moving ? "moving" : "static";
```

**Review by takayuki5168 — APPROVED** [review]

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp (hunk 4 above)
Comment (danielsanchezaran) [reply planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp]: Done!

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp (hunk 3 above)
Comment (danielsanchezaran) [reply planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp]: Done!

**Pull request merged**